core/src/main/resources/lib/layout/breadcrumbBar.jelly

```diff
@@ -44,8 +44,14 @@ THE SOFTWARE.
               <j:forEach var="anc" items="${request.ancestors}">
                 <j:if test="${h.isModel(anc.object) and anc.prev.url!=anc.url}">
                   <li class="item">
+                    <!-- Workaround to set desired name to first breadcrumb (instead of Jenkins) -->
                     <a href="${anc.url}/" class="${h.isModelWithContextMenu(anc.object)?'model-link':null} inside">
-                      ${anc.object.displayName}
+                      <j:if test="${anc.object.displayName == 'Jenkins'}">
```

Comment:
```suggestion
                      <j:if test="${anc.object == app}">
```

Limit to the actual root element. Untested.

Reply:
Applied and tested. Works great!